river/LayoutManager.zig

```diff
@@ -0,0 +1,378 @@
+                }
+
+                // Apply proposed layout to views and start transaction
+                var it = ViewStack(View).iter(self.output.views.first, .forward, self.output.pending.tags, Output.arrangeFilter);
+                while (it.next()) |view| {
+                    view.pending.box = self.output.layout_demand.?.popViewDimensions() orelse unreachable;
+                    view.applyConstraints();
+                }
+                self.output.pending.layout = true;
+
+                // TODO layout name?
+            },
+        }
+    }
+};
+
+/// This object encapsulates a zriver.LayoutManagerV1 object bound by a client
+const LayoutManager = struct {
+    parent: *Self,
+    layout_manager: *zriver.LayoutManagerV1,
+
+    /// Initialize a connection
+    pub fn init(self: *LayoutManager, parent: *Self, client: *wl.Client, version: u32, id: u32) !void {
+        self.layout_manager = zriver.LayoutManagerV1.create(client, version, id) catch {
+            self.layout_manager.postNoMemory();
+            log.crit(.river_layout, "out of memory", .{});
+            return Error.BindFailed;
+        };
+        self.parent = parent;
+        self.layout_manager.setHandler(*LayoutManager, handleRequest, null, self);
+    }
+
+    pub fn deinit(self: *LayoutManager) void {
+        self.layout_manager.destroy();
+    }
+
+    fn handleRequest(layout_manager: *zriver.LayoutManagerV1, request: zriver.LayoutManagerV1.Request, self: *LayoutManager) void {
+        switch (request) {
+            .destroy => {
+                self.deinit();
+
+                // Remove this client from the global clients list
+                const node = @fieldParentPtr(std.TailQueue(LayoutManager).Node, "data", self);
+                self.parent.layout_managers.remove(node);
+                util.gpa.destroy(node);
+            },
+
+            .get_river_layout => |req| {
+                // Ignore if the output is inert
+                const wlr_output = wlr.Output.fromWlOutput(req.output) orelse return;
+                const output = @intToPtr(*Output, wlr_output.data);
+
+                log.debug(.layout_manager, "bind layout '{}' on output '{}'", .{ req.namespace, output.wlr_output.name });
+
+                const node = util.gpa.create(std.TailQueue(Layout).Node) catch {
+                    layout_manager.getClient().postNoMemory();
+                    log.crit(.river_layout, "out of memory", .{});
+                    return;
+                };
+                node.data.init(req.namespace, output, layout_manager.getClient(), 1, req.id) catch {
+                    layout_manager.getClient().postNoMemory();
+                    log.crit(.river_layout, "out of memory", .{});
+                    util.gpa.destroy(node);
+                    return;
+                };
+                output.layouts.append(node);
+            },
+        }
+    }
+};
+
+global: *wl.Global,
+server_destroy: wl.Listener(*wl.Server) = undefined,
```

Comment:
```suggestion
server_destroy: wl.Listener(*wl.Server) = wl.Listener(*wl.Server).init(handleServerDestroy),
```
(and remove the `setNotify()` call below)